nexus/db-model/src/affinity.rs

```diff
@@ -257,3 +258,30 @@ impl From<AntiAffinityGroupInstanceMembership>
         Self::Instance(member.instance_id.into())
     }
 }
+
+#[derive(Queryable, Insertable, Clone, Debug, Selectable)]
+#[diesel(table_name = anti_affinity_group_affinity_membership)]
+pub struct AntiAffinityGroupAffinityMembership {
+    pub anti_affinity_group_id: DbTypedUuid<AntiAffinityGroupKind>,
+    pub affinity_group_id: DbTypedUuid<AffinityGroupKind>,
+}
+
+impl AntiAffinityGroupAffinityMembership {
+    pub fn new(
+        anti_affinity_group_id: AntiAffinityGroupUuid,
+        affinity_group_id: AffinityGroupUuid,
+    ) -> Self {
+        Self {
+            anti_affinity_group_id: anti_affinity_group_id.into(),
+            affinity_group_id: affinity_group_id.into(),
+        }
+    }
+}
```

Comment:
this is maybe a bit goofy of me, but for stuff like this where there's a clear directional relationship (the anti-affinity group is a member of the affinity group),i _kinda_ wonder whether we ought to have a method on `AntiAffinityGroupUuid` be the constructor, so you can write
```rust
let membership = anti_affinity_group_id.is_member_of(affinity_group_id);
```
and have that relationship be clearly stated at the call site? On the other hand, this might just be annoying and deviate from the convention for constructors for no real reason. It's the type of thing I would have reached for in my misspent youth[^1] which makes me kind of leery of this suggestion...

[^1] i.e. back when i wrote Scala